executor/aggregate.go

```diff
@@ -375,9 +371,11 @@ func (w *HashAggPartialWorker) run(ctx sessionctx.Context, waitGroup *sync.WaitG
 			w.shuffleIntermData(sc, finalConcurrency)
 		}
 		w.memTracker.Consume(-w.chk.MemoryUsage())
-		waitGroup.Done()
```

Comment:
why remove this?

Reply:
In the original scheme, it needs to be placed in the outer layer, otherwise there will be race conditions. Will be put back into the run() function in the next commit